spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,381 @@
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return greaterThanOrEqual(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case GT:
+          if (rightChild(predicate) instanceof Literal) {
+            return greaterThan(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return lessThan(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case GT_EQ:
+          if (rightChild(predicate) instanceof Literal) {
+            return greaterThanOrEqual(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return lessThanOrEqual(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case EQ: // used for both eq and null-safe-eq
+          Object value;
+          String attributeName;
+          if (rightChild(predicate) instanceof Literal) {
+            attributeName = toColumnName(leftChild(predicate));
+            value = convertUTF8StringIfNecessary(rightChild(predicate));
+          } else {
+            attributeName = toColumnName(rightChild(predicate));
+            value = convertUTF8StringIfNecessary(leftChild(predicate));
+          }
+
+          if (predicate.name().equals("=")) {
+            // comparison with null in normal equality is always null. this is probably a mistake.
+            Preconditions.checkNotNull(
+                value, "Expression is always false (eq is not null-safe): %s", predicate);
+            return handleEqual(attributeName, value);
+          } else { // "<=>"
+            if (value == null) {
+              return isNull(attributeName);
+            } else {
+              return handleEqual(attributeName, value);
+            }
+          }
+
+        case IN:
+          return in(
+              toColumnName(leftChild(predicate)),
+              Arrays.stream(predicate.children())
+                  .skip(1)
+                  .map(val -> convertUTF8StringIfNecessary(((Literal) val)))
+                  .filter(Objects::nonNull)
+                  .collect(Collectors.toList()));
+
+        case NOT:
+          Not notFilter = (Not) predicate;
+          Predicate childFilter = notFilter.child();
+          if (childFilter.name().equals("IN")) {
+            // infer an extra notNull predicate for Spark NOT IN filters
+            // as Iceberg expressions don't follow the 3-value SQL boolean logic
+            // col NOT IN (1, 2) in Spark is equivalent to notNull(col) && notIn(col, 1, 2) in
```

Comment:
nit: What about replacing `equivalent` with `equal` so that the last line of the comment would fit on one line?

Reply:
Fixed. Thanks!